### PR TITLE
feat: KT-45842 sealed class inheritance in MPP

### PR DESCRIPTION
**Pull request by p7nov**
Update doc as per https://youtrack.jetbrains.com/issue/KT-45842

### CODE REVIEW

**Review by wild-lynx — APPROVED** [review]

File: docs/topics/sealed-classes.md [hunk 1]
```diff
@@ -50,6 +50,18 @@ Subclasses of sealed classes must have a proper qualified name. They can't be lo
 >
 {type="note"}
 
+### Inheritance in multiplatform projects
+
+In [multiplatform projects](mpp-intro.md), there comes one more restriction: direct subclasses of sealed classes must
```
Comment (koshachy) [inline docs/topics/sealed-classes.md:55]: I prefer more simplified wording in such cases, like "There is an inheritance restriction in multiplatform projects:", but yours is also good!

File: docs/topics/sealed-classes.md [hunk 2]
```diff
@@ -50,6 +50,18 @@ Subclasses of sealed classes must have a proper qualified name. They can't be lo
...
+reside in the same source set. It applies to sealed classes without the [`expect`/`actual`](mpp-connect-to-apis.md) modifiers.
+
+A sealed class can be declared as `expect` in a common source set and have `actual` implementations in platform source sets.
+In this case, both `expect` and `actual` classes can have subclasses in their source sets. Moreover, if you use [hierarchical structure](mpp-share-on-platforms.md#share-code-on-similar-platforms),
```
Comment (koshachy) [inline docs/topics/sealed-classes.md:59]:
```suggestion
In this case, both `expect` and `actual` classes can have subclasses in their source sets. Moreover, if you use a [hierarchical structure](mpp-share-on-platforms.md#share-code-on-similar-platforms),
```

File: docs/topics/sealed-classes.md [hunk 3]
```diff
@@ -50,6 +50,18 @@ Subclasses of sealed classes must have a proper qualified name. They can't be lo
 >
 {type="note"}
 
+### Inheritance in multiplatform projects
+
+In [multiplatform projects](mpp-intro.md), there comes one more restriction: direct subclasses of sealed classes must
+reside in the same source set. It applies to sealed classes without the [`expect`/`actual`](mpp-connect-to-apis.md) modifiers.
```
Comment (koshachy) [inline docs/topics/sealed-classes.md:56]: Maybe a conjunction suits here better than the backslash sign.

File: docs/topics/sealed-classes.md [hunk 4]
```diff
@@ -50,6 +50,18 @@ Subclasses of sealed classes must have a proper qualified name. They can't be lo
...
+
+A sealed class can be declared as `expect` in a common source set and have `actual` implementations in platform source sets.
```
Comment (koshachy) [inline docs/topics/sealed-classes.md:58]: The second paragraph is a bit unclear for me — it seems to be a little out of touch with the beginning, where the restriction is explained.
I cannot offer a better structure here yet, but I think we can use this one (I mean yours) and return to it in the future. WDYT?

File: docs/topics/sealed-classes.md (hunk 2 above)
Comment (koshachy) [reply docs/topics/sealed-classes.md:59]: You can omit this if it was suggested by the copy editors team.

File: docs/topics/sealed-classes.md (hunk 4 above)
Comment (p7nov) [reply docs/topics/sealed-classes.md]: I've changed the intro a bit. Not sure this is worth returning to, let's see if we have complaints.

**Pull request merged**